geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/io/SimpleFeatureParquetSchema.scala

```diff
@@ -30,19 +30,20 @@ import org.locationtech.geomesa.utils.text.StringSerialization
   * @param sft simple feature type
   * @param schema parquet message schema
   */
-case class SimpleFeatureParquetSchema(sft: SimpleFeatureType, schema: MessageType) {
+case class SimpleFeatureParquetSchema(sft: SimpleFeatureType, schema: MessageType, version: Integer = CurrentSchemaVersion) {
 
-  import SimpleFeatureParquetSchema.{CurrentSchemaVersion, SchemaVersionKey}
+  import SimpleFeatureParquetSchema.{CurrentSchemaVersion, GeoParquetSchemaKey, SchemaVersionKey}
 
   import scala.collection.JavaConverters._
 
   /**
     * Parquet file metadata
     */
-  lazy val metadata: java.util.Map[String, String] = Map(
+  lazy val metadata: java.util.Map[String, String] = scala.collection.mutable.HashMap(
```

Comment:
we probably don't want this to be a mutable map, especially since it's not a private variable. having mutable state makes things harder to reason about, since you don't know if someone has modified it out from under you.